utils/optiwebp.py

```diff
@@ -0,0 +1,160 @@
+
+args = argparse.ArgumentParser()
+
+args.add_argument("--minpercent", default=90, help="The new file size must be reduced by to least this percentage compared to the original. Default 90%% (10%% reduction).")
+args.add_argument("--nodryrun", action="store_true", help="Whether to replace the original file with the converted one if its size is reduced enough. Does not replace by default.")
+args.add_argument("--quality", default="90", help="The quality to use for lossy compression. Defaults to 90.")
+args.add_argument("--repo", default=os.getcwd(), help="The root of the Wesnoth repository. Defaults to the current directory.")
+args.add_argument("--tempdir", default=os.path.join(os.getcwd(), "temp_webp"), help="The root of the Wesnoth repository. Defaults to 'temp_webp/' in the current directory.")
+
+options = args.parse_args()
+
+# test we're in a repo root
+if not os.path.isfile(os.path.join(options.repo, "changelog.md")):
+    print("Didn't find changelog.md so assuming this isn't the repo root, exiting.")
+    sys.exit(1)
+
+# test cwebp presence
+if subprocess.run(["cwebp", "-version"], stdout=subprocess.DEVNULL).returncode != 0:
+    print("cwebp executable not found in PATH, exiting.")
+    sys.exit(1)
+
+image_dirs = [
+    "data/core/images/",
+    "data/campaigns/",
+    "images/"
+]
+
+results_dict = {}
+lossy_args = ["cwebp", "-mt", "-m", "6", "-q", options.quality, "-alpha_q", "100"]
+lossless_args = ["cwebp", "-mt", "-z", "9", "-lossless", "-exact"]
+
+os.makedirs(options.tempdir, exist_ok=True)
+
+start = time.time()
+count = 0
+
+for image_dir in image_dirs:
+    print("Processing "+image_dir)
+    for image_root, _, files in os.walk(os.path.join(options.repo, image_dir)):
+        for filename in files:
+            count += 1
+            if count % 100 == 0:
+                print("Processing file {count}".format(count=count))
+
+            filetype = Path(filename).suffix
+            subprocess_args = []
+            if filetype == ".png":
+                subprocess_args = lossless_args
+            elif filetype == ".webp":
+                with open(os.path.join(image_root, filename), "rb") as img:
+                    # discard first 12 bytes of WebP "magic number" and file size
+                    img.read(12)
+                    # read type of WebP file
+                    # VP8L = lossless
+                    # VP8 = lossy
+                    # VP8X = can either, so assume lossless (better to try to go from lossy->lossless than the reverse)
+                    webp_type = img.read(4)
+                    if webp_type == b"VP8L" or webp_type == b"VP8X":
+                        subprocess_args = lossless_args
+                    elif webp_type == b"VP8 ":
+                        subprocess_args = lossy_args
+                    else:
+                        print(filename, "is not a valid WebP file", file=sys.stderr)
+                        continue
+            elif filetype == ".jpg":
```

Comment:
JPEG files can also have the `.jpeg` extension, so you should add `or filetype == ".jpeg"` (or you can use `elif filetype in (".jpg", ".jpeg"):`.

Reply:
Done.